src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -61,11 +63,27 @@ public async Task<Compilation> CompileAsync(string inputPath, bool skipRestore)
                 }
             }
 
-            var compilation = new Compilation(this.invocationContext.ResourceTypeProvider, sourceFileGrouping);
+            try
+            {
+                var compilation = new Compilation(this.invocationContext.ResourceTypeProvider, sourceFileGrouping);
```

Comment:
>Compilation [](http://example.com/codeflow?start=38&length=11)

Wouldn't this also happen in the language server? I think we should make it so the compilation doesn't throw when the config file is malformed.

Reply:
Basically the reasoning is that the user can get much of the functionality even with a malformed config file, so the compilation shouldn't just give up.

Reply:
Like I mentioned in the below comment, Compilation needs a valid config. If we encounter an invalid config, should we surface exception to user, create default disabled linter config and proceed with creating compilation?

Reply:
Created below issue to track the issue on language server side:
https://github.com/Azure/bicep/issues/4471

Reply:
Made changes to write error message and go ahead with compilation creation:
https://github.com/Azure/bicep/commit/699af2a087c920c1002078f3d05e2016783c7f07